akka-stream-tests/src/test/scala/akka/stream/impl/fusing/InterpreterSpec.scala

```diff
@@ -569,39 +569,69 @@ class InterpreterSpec extends AkkaSpec with GraphInterpreterSpecKit {
 
   }
 
-  private[akka] case class Doubler[T]() extends PushPullStage[T, T] {
-    var oneMore: Boolean = false
-    var lastElem: T = _
-
-    override def onPush(elem: T, ctx: Context[T]): SyncDirective = {
-      lastElem = elem
-      oneMore = true
-      ctx.push(elem)
-    }
+  private[akka] final case class Doubler[T]() extends GraphStage[FlowShape[T, T]] {
```

Comment:
Ah, it's basically just for the apply method, yeah, I guess that is ok then (but doesn't make sense in general)